### ISSUE TITLE
linter: getter-return false positive

### ISSUE THREAD

**Issue by Boshen** [oxc-project/oxc#3935]
```javascript
  get steps() {
    const steps = [];

    const gradient = colorGradient('#3D68FF', '#8845FF', this.maxStepsNumber);

    for (let i = 0; i < this.maxStepsNumber; i++) {
      steps.push({
        stepnum: i + 1,
        status: i <= this.currentStepIndex ? 'active' : '',
        background: htmlSafe(`background: ${gradient[i]};`),
      });
    }

    return steps;
  }
```

https://github.com/1024pix/pix/blob/1352bd8d7f6070f1ff8da79867f543c1c1926e59/mon-pix/app/components/progress-bar.js#L29-L43
